model.py

```diff
@@ -289,6 +346,14 @@ def _run(self, inputs, labels=None, device='CPU', device_ids=None):
 
     def _make_program(self, inputs):
         prog = self._orig_prog.clone()
+        # change inputs to the same var in cloned program
+        inputs = fluid.layers.utils.map_structure(
+            lambda var: prog.global_block().var(var.name), inputs)
+        # prune unraleted ops in test program, mainly for ops inserted
+        # by learning rate scheduling
+        if self.mode != 'train':
+            for op in list(prog.global_block().ops):
+                prog.global_block()._remove_op(0)
```

Comment:
quite hard to get one's head around this part, need more clarification in comments, If I am understanding it correctly:
- the learning rate  schedulers insert vars when constructed, which happens before first run
- this line remove all ops in the program, since it is before `forward`, the program contains only optimizer related vars
- one thing I am not so sure about: if the LR scheduler vars were not to be removed, how will it affect the training process? (the counter will increment even without`minimize` call?)

Reply:
Done.  Update the comments. 

> if the LR scheduler vars were not to be removed, how will it affect the training process? (the counter will increment even withoutminimize call?)

As you said, the counter will increment even without minimize call . And we can `print(fluid.default_main_program())`  to inspect the ops to increase global step var after `model.prepare`.

Reply:
okay, I see